packages/sitecore-jss/src/personalization/layout-fragment-service.ts

```diff
@@ -0,0 +1,95 @@
+import { ComponentRendering } from '../layout/models';
+import { GraphQLRequestClient } from './../graphql-request-client';
+import debug from './../debug';
+
+export interface LayoutFragmentData {
+  fragment: ComponentRendering;
+}
+
+export interface LayoutFragmentService {
+  fetchLayoutFragmentData(
+    routePath: string,
+    language: string,
+    renderingId: string,
+    ruleId: string
+  ): Promise<LayoutFragmentData>;
+}
+
+export type GraphQLLayoutFragmentServiceConfig = {
+  /**
+   * Your Graphql endpoint
+   */
+  endpoint: string;
+  /**
+   * The JSS application name
+   */
+  siteName: string;
+  /**
+   * The API key to use for authentication
+   */
+  apiKey: string;
```

Comment:
Is apiKey required?
Is it really used for authentication?

Reply:
The comment is exactly the same as for other GraphQL services. I suppose in case of exp edge this key is used for authentication somehow